File: Problems/Construct the Minimum Bitwise Array I.py
class Solution:
    def minBitwiseArray(self, nums):
        def helper(num):
            if  num % 4 == 1:           
                return num - 1 
            if num % 4 == 3:         
                tmp = num
                for i in range(num.bit_length()):
                    tmp //= 2
                    if not tmp % 2:
                        break
                return num - (1<<i)     
            return -1                   
        return map(helper,nums)

File: Problems/Construct the Minimum Bitwise Array I.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        vector<int> ans;

        for (auto& num : nums) {
            if (num == 2) {
                ans.push_back(-1);
            } 
            else {
                for (char i = 10; i >= 0; i--) {
                    if((num & (1 << i)) != 0){
                        int temp = num & ~(1 << i);
                        if((temp++ | temp) == num){
                            ans.push_back(temp - 1);
                            break;
                        }
                    }
                }
            }
        }

        return ans;
    }
};
